runtime/gc_glue_java/MetronomeDelegate.hpp

```diff
@@ -157,10 +166,374 @@ class MM_MetronomeDelegate : public MM_BaseNonVirtual
 	void acquireExclusiveVMAccess(MM_EnvironmentBase *env, bool waitRequired);
 	void releaseExclusiveVMAccess(MM_EnvironmentBase *env, bool releaseRequired);
 
+	void markLiveObjects(MM_EnvironmentRealtime *env);
+#if defined(J9VM_GC_DYNAMIC_CLASS_UNLOADING)
+	MMINLINE void markClassOfObject(MM_EnvironmentRealtime *env, J9Object *objectPtr)
+	{
+		markClassNoCheck(env, J9GC_J9OBJECT_CLAZZ(objectPtr));
+	}
+	MMINLINE bool markClassNoCheck(MM_EnvironmentRealtime *env, J9Class *clazz)
+	{
+		bool result = false;
+		if (J9_ARE_ANY_BITS_SET(J9CLASS_EXTENDED_FLAGS(clazz), J9ClassIsAnonymous)) {
+			/*
+			 * If class is anonymous it's classloader will not be rescanned
+			 * so class object should be marked directly
+			 */
+			result = _markingScheme->markObject(env, clazz->classObject);
+		} else {
+			result = _markingScheme->markObject(env, clazz->classLoader->classLoaderObject);
+		}
+		return result;
+	}
+	bool markClass(MM_EnvironmentRealtime *env, J9Class *objectPtr);
+#endif /* defined(J9VM_GC_DYNAMIC_CLASS_UNLOADING) */
+
+	MMINLINE uintptr_t scanPointerArraylet(MM_EnvironmentRealtime *env, fomrobject_t *arraylet)
+	{
+		fj9object_t *startScanPtr = arraylet;
+		fj9object_t *endScanPtr = startScanPtr + (_javaVM->arrayletLeafSize / sizeof(fj9object_t));
+		return scanPointerRange(env, startScanPtr, endScanPtr);
```

Comment:
Style: mixing of `fomrobject_t` and `fj9object_t` in the same function looks inconsistent. Considering transition to OMR types I would replace  `fj9object_t` to `fomrobject_t` in this function